iXBRLViewerPlugin/__init__.py

```diff
@@ -143,8 +164,42 @@ def setLaunchIXBRLViewer(self, *args):
 
 def guiRun(cntlr, modelXbrl, attach, *args, **kwargs):
     """ run iXBRL Viewer using GUI interactions for a single instance or testcases """
-    if cntlr.hasGui and cntlr.launchIXBRLViewer.get() and modelXbrl.modelDocument.type in (Type.INLINEXBRL, Type.INLINEXBRLDOCUMENTSET):
-        launchLocalViewer(cntlr, modelXbrl)
+    class iXBRLViewerLocalViewer(LocalViewer):
+        # plugin-specific local file handler
+        def getLocalFile(self, file, relpath, request):
```

Comment:
Why did you do this? Why not pull this function out into its own function with documentation?

Reply:
I can pull the class out if you want, but I just moved some code. `getLocalFile` is an override to `LocalViewer.getLocalFile`